packages/packages/core/editor-canvas/src/components/elements-overlays.tsx

```diff
@@ -21,14 +43,26 @@ export function ElementsOverlays() {
 
 	return (
 		isActive &&
-		elements.map( ( [ id, element ] ) => (
-			<ElementOverlay key={ id } id={ id } element={ element } isSelected={ selected.element?.id === id } />
-		) )
+		elements.map( ( [ id, element ] ) => {
+			const isSelected = selected.element?.id === id;
+
+			return overlayRegistry
+				.filter( ( overlay ) => overlay.filter( { id, element, isSelected } ) )
+				.map( ( overlay, index ) => {
+					const OverlayComponent = overlay.component;
+					return (
+						<OverlayComponent
+							key={ `${ id }-${ index }` }
+							id={ id }
+							element={ element }
+							isSelected={ isSelected }
+						/>
+					);
+				} );
```

Comment:
map method can return undefined in React render context
and since filter is also an array method, I would name it `shouldRender` to avoid confusion
```suggestion
				.map( ( overlay, index ) => {
					const OverlayComponent = overlay.component;

					return ( overlay.filter( { id, element, isSelected } ) ) &&
						<OverlayComponent
							key={ `${ id }-${ index }` }
							id={ id }
							element={ element }
							isSelected={ isSelected }
						/>
					);
				} );
```